packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts

```diff
@@ -25,7 +25,7 @@ export function isValidCollectionPathImplementation(
 
     try {
       assert(opener, `{`, ``) // This is a noop because of the opening check, but here for posterity
-      assert(model, /^[A-Z][a-zA-Z]+$/, errorMessage(part))
+      assert(model, /^[a-zA-Z_][a-zA-Z0-9_]+$/, errorMessage(part))
       assert(field, /^[a-zA-Z_()]+$/, errorMessage(part))
```

Comment:
Shouldn't this check also get `\d`, at least, like above?

Reply:
Yep, the same https://spec.graphql.org/draft/#sec-Names also applies to fields